File: src/main/java/org/shark/example/controller/redis/RedisListController.java
package org.shark.example.controller.redis;

import com.google.gson.Gson;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.controller.redis.pojo.KeyStringDto;
import org.shark.example.service.base.pojo.ResponseDto;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequiredArgsConstructor
@RestController
@RequestMapping("/redis/list")
@Slf4j
public class RedisListController {

    private final StringRedisTemplate stringRedisTemplate;

    @PostMapping
    public ResponseDto<Long> set(
            @RequestBody KeyStringDto setStringOptRedisKeyDto) {
        Long count = stringRedisTemplate.opsForList().rightPush(setStringOptRedisKeyDto.getKey(), setStringOptRedisKeyDto.getValue());
        return ResponseDto.<Long>builder().data(count).status(true).build();
    }

    @GetMapping("/key/{key}")
    public ResponseDto<List<String>> get(@PathVariable String key) {
        Long length = stringRedisTemplate.opsForList().size(key);
        List<String> list = stringRedisTemplate.opsForList().range(key, 0, length);
        log.info(new Gson().toJson(list));
        return ResponseDto.<List<String>>builder().data(list).status(true).build();
    }
}

File: src/main/kotlin/org/shark/example/controller/redis/RedisListController.kt
package org.shark.example.controller.redis

import com.google.gson.Gson
import org.shark.example.controller.redis.pojo.KeyStringDto
import org.shark.example.service.base.pojo.ResponseDto
import org.slf4j.LoggerFactory
import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/redis/list")
class RedisListController(
    private val stringRedisTemplate: StringRedisTemplate
) {
    private val log = LoggerFactory.getLogger(RedisListController::class.java)

    @PostMapping
    fun set(@RequestBody setStringOptRedisKeyDto: KeyStringDto): ResponseDto<Long> {
        val count = if (setStringOptRedisKeyDto.key != null && setStringOptRedisKeyDto.value != null) {
            stringRedisTemplate.opsForList().rightPush(setStringOptRedisKeyDto.key, setStringOptRedisKeyDto.value)
        } else {
            0L
        }
        return ResponseDto(data = count, status = true)
    }

    @GetMapping("/key/{key}")
    fun get(@PathVariable key: String): ResponseDto<List<String>> {
        val length = stringRedisTemplate.opsForList().size(key) ?: 0
        val list = stringRedisTemplate.opsForList().range(key, 0, length)
        log.info(Gson().toJson(list))
        return ResponseDto(data = list, status = true)
    }
}
